includes/subscriptions/class-wc-payments-subscription-service.php

```diff
@@ -269,9 +269,15 @@ public static function format_item_price_data( string $currency, string $wcpay_p
 		$data = [
 			'currency'            => $currency,
 			'product'             => $wcpay_product_id,
-			'unit_amount_decimal' => round( $unit_amount, wc_get_rounding_precision() ) * 100,
+			// We cannot use WC_Payments_Utils::prepare_amount() here because it returns an int but 'unit_amount_decimal' supports multiple decimal places even though it is cents (fractions of a cent).
+			'unit_amount_decimal' => round( $unit_amount, wc_get_rounding_precision() ),
```

Comment:
@james-allan I think this approach is still too hacky and might cause troubles in the future, so I'd suggest refactoring it.

1. Extend signature of prepare_amount to accept precision as optional third argument (0 by default).
2. Use it as: `WC_Payments_Utils::prepare_amount( $unit_amount,  $currency, wc_get_rounding_precision() )`
3. Update return result from prepare_amount: 
```php
return $precision > 0 ? 
    round( (float) $amount * $conversion_rate, $precision ) :
    (int) round( (float) $amount * $conversion_rate, $precision ); 
```

Reply:
One caveat I'm not fully sure about is  when `wc_get_rounding_precision()` returns 0. Is it ok to pass `unit_amount_decimal` as int? If not you could make it explicit as (float).
`'unit_amount_decimal' => (float) WC_Payments_Utils::prepare_amount( $unit_amount, $currency, wc_get_rounding_precision() )`